arithmetic/database.py

```
@@ -0,0 +1,173 @@
+import csv
+import os
+from contextlib import contextmanager
+
+
+class ArithmeticDatabase(object):
+    """
+    A simple database backed by a csv file to contain arithmetic operations
+    with two integers and the corresponding answer
+    """
+    headers = ['operator', 'int1', 'int2', 'answer']
+    valid_operators = ['add', 'subtract', 'multiply', 'divide']
+    buffer_size = 1024
+
+    def __init__(self, filepath, use_existing=False):
+        """
+        Initialize the csv file used as the basis for the database. Also write
+        the headers to the file
+
+        Args:
+            filepath (str): path for the csv file
+            use_existing (bool): if True, initialize with an existing file
+
+        Raises:
+            Exception: if not `use_existing` and a file with `filepath` exists
+        """
+        self.filepath = filepath
+        if os.path.exists(filepath):
+            if not use_existing:
+                return Exception('Cannot initialize ArithmeticDatabase; file %s already exists' % filepath)
```

Should be 'raise' instead of 'return'?